test/fw/ptl/utils/pbs_dshutils.py

```diff
@@ -1919,13 +1944,20 @@ def create_temp_file(self, hostname=None, suffix='', prefix='PtlPbs',
             else:
                 os.write(fd, body)
         os.close(fd)
+
+        if self.get_platform() == 'shasta' and not hostname and asuser:
+            asuser = PbsUser.get_user(asuser)
+            if asuser.host:
+                self.logger.info('set hostname to userhost')
+                hostname = asuser.host
+
         # if temp file to be created on remote host
         if not self.is_localhost(hostname):
             if asuser is not None:
                 # by default mkstemp creates file with 0600 permission
                 # to create file as different user first change the file
                 # permission to 0644 so that other user has read permission
-                self.chmod(hostname, tmpfile, mode=0644)
+                self.chmod(tmpfile, mode=0644)
```

Comment:
why did you remove the hostname arg?

Reply:
It was a bug, PTL creates a file on the local host, then chmods the file on the remote host? The file hasn't been copied yet.

Reply:
ok, but now you are passing the file path 'tmpfile' into the 'hostname' argument of the function, you should change this to path=tmpfile

Reply:
> ok, but now you are passing the file path 'tmpfile' into the 'hostname' argument of the function, you should change this to path=tmpfile

I think you missed this comment

Reply:
Ah, I missed that. Thanks!